hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/Spark24HoodieParquetReader.scala

```diff
@@ -0,0 +1,222 @@
+   * @param sqlConf    the [[SQLConf]] used for the read
+   * @param options    passed as a param to the file format
+   * @param hadoopConf some configs will be set for the hadoopConf
+   * @return map of properties needed for reading a parquet file
+   */
+  def getPropsForReadingParquet(vectorized: Boolean,
+                                sqlConf: SQLConf,
+                                options: Map[String, String],
+                                hadoopConf: Configuration): Map[String, String] = {
+    //set hadoopconf
+    hadoopConf.set(ParquetInputFormat.READ_SUPPORT_CLASS, classOf[ParquetReadSupport].getName)
+    hadoopConf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, sqlConf.sessionLocalTimeZone)
+    hadoopConf.setBoolean(SQLConf.CASE_SENSITIVE.key, sqlConf.caseSensitiveAnalysis)
+    hadoopConf.setBoolean(SQLConf.PARQUET_BINARY_AS_STRING.key, sqlConf.isParquetBinaryAsString)
+    hadoopConf.setBoolean(SQLConf.PARQUET_INT96_AS_TIMESTAMP.key, sqlConf.isParquetINT96AsTimestamp)
+
+    Map(
+      "enableVectorizedReader" -> vectorized.toString,
+      "enableParquetFilterPushDown" -> sqlConf.parquetFilterPushDown.toString,
+      "pushDownDate" -> sqlConf.parquetFilterPushDownDate.toString,
+      "pushDownTimestamp" -> sqlConf.parquetFilterPushDownTimestamp.toString,
+      "pushDownDecimal" -> sqlConf.parquetFilterPushDownDecimal.toString,
+      "pushDownInFilterThreshold" -> sqlConf.parquetFilterPushDownInFilterThreshold.toString,
+      "pushDownStringStartWith" -> sqlConf.parquetFilterPushDownStringStartWith.toString,
+      "isCaseSensitive" -> sqlConf.caseSensitiveAnalysis.toString,
+      "timestampConversion" -> sqlConf.isParquetINT96TimestampConversion.toString,
+      "enableOffHeapColumnVector" -> sqlConf.offHeapColumnVectorEnabled.toString,
+      "capacity" -> sqlConf.parquetVectorizedReaderBatchSize.toString,
+      "returningBatch" -> sqlConf.parquetVectorizedReaderEnabled.toString,
+      "enableRecordFilter" -> sqlConf.parquetRecordFilterEnabled.toString,
+      "timeZoneId" -> sqlConf.sessionLocalTimeZone
+    )
+  }
+
+  /**
+   * Read an individual parquet file
+   * Code from ParquetFileFormat#buildReaderWithPartitionValues from Spark v2.4.8 adapted here
+   *
+   * @param file            parquet file to read
+   * @param requiredSchema  desired output schema of the data
+   * @param partitionSchema schema of the partition columns. Partition values will be appended to the end of every row
+   * @param filters         filters for data skipping. Not guaranteed to be used; the spark plan will also apply the filters.
+   * @param sharedConf      the hadoop conf
+   * @param extraProps      properties generated by [[getPropsForReadingParquet]] that are needed for reading
+   * @return iterator of rows read from the file output type says [[InternalRow]] but could be [[ColumnarBatch]]
+   */
+  def readParquetFile(file: PartitionedFile,
```

Comment:
Can you add tests on `getPropsForReadingParquet` and `readParquetFile`?

Reply:
Let's still add simple test for each Spark version to make sure the new Parquet reader can read a parquet file.

Reply:
Let's still add a simple test for each Spark version to make sure the new Parquet reader can read a parquet file.